sample-geovelocity/siddhi-execution-geovelocity/src/main/java/org/wso2/extension/siddhi/execution/geovelocity/GetLoginBehaviourRisk.java

```diff
@@ -0,0 +1,177 @@
+                        "insert into outputStream;")
+)
+
+public class GetLoginBehaviourRisk extends FunctionExecutor {
+
+    private static final String DEFAULT_GEOVELOCITY_RESOLVER_CLASSNAME =
+            "org.wso2.extension.siddhi.execution.geovelocity.internal.impl.DefaultDBBasedGeoVelocityDataResolver";
+    private static GeoVelocityDataResolver geoVelocityDataResolverImpl;
+
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                        SiddhiAppContext siddhiAppContext) {
+
+        initializeExtensionConfigs(configReader);
+        if (attributeExpressionExecutors.length != 3) {
+            throw new SiddhiAppValidationException("Invalid no of arguments passed to geovelocity:loginbehaviourrisk" +
+                    " function, required 3, but found " + attributeExpressionExecutors.length);
+        }
+        Attribute.Type attributeType = attributeExpressionExecutors[0].getReturnType();
+        if (attributeType != Attribute.Type.STRING) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for first argument username of " +
+                    "geovelocity:loginbehaviourbasedrisk() function, required " + Attribute.Type.STRING + ", but found "
+                    + attributeType.toString());
+        }
+        Attribute.Type secondAttributeType = attributeExpressionExecutors[1].getReturnType();
+        if (secondAttributeType != Attribute.Type.STRING) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for second argument city of " +
+                    "geovelocity:loginbehaviourbasedrisk() function, required " + Attribute.Type.STRING + ", but found "
+                    + secondAttributeType.toString());
+        }
+        Attribute.Type thirdaAttributeType = attributeExpressionExecutors[2].getReturnType();
+        if (thirdaAttributeType != Attribute.Type.LONG) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for third argument " +
+                    "currentlogintime of geovelocity:loginbehaviourbasedrisk() function, required "
+                    + Attribute.Type.LONG + ", but found " + thirdaAttributeType.toString());
+        }
+    }
+
+    @Override
+    protected Object execute(Object[] data) {
+
+        String username = data[0].toString();
+        String city = data[1].toString();
+        Long currentLoginTime = Long.parseLong(data[2].toString());
+        GeoVelocityData geoVelocityData = geoVelocityDataResolverImpl.getGeoVelocityInfo(username, city);
+        Long lastLoginTime = geoVelocityData.getLoginBehaviourBasedRisk();
+        double risk;
+        if (lastLoginTime != 0L) {
+            //Time difference is converted to days.
+            long timeDifference = TimeUnit.MILLISECONDS.toDays(currentLoginTime - lastLoginTime);
+            //Risk was calculated by using the formula of (1 - e^(-t)). t = Time difference.
```

Comment:
Keep a space when starting the comment

Reply:
Fixed in https://github.com/wso2/samples-is/pull/265